packages/viewer/src/router/storeSync/params/layers.param.ts

```diff
@@ -0,0 +1,440 @@
+
+            if (layerFeatureRequests) {
+                featuresRequests.push(...layerFeatureRequests)
+            }
+
+            if (layerObject) {
+                if (layerObject.type === LayerType.KML && (layerObject as KMLLayer).adminId) {
+                    drawingStore.toggleDrawingOverlay(
+                        {
+                            show: true,
+                        },
+                        STORE_DISPATCHER_ROUTER_PLUGIN
+                    )
+                    drawingStore.setIsVisitWithAdminId(true, STORE_DISPATCHER_ROUTER_PLUGIN)
+                }
+                log.debug({
+                    title: 'Layers URL param / dispatchLayersFromUrlIntoStore',
+                    titleColor: LogPreDefinedColor.Sky,
+                    messages: [`Add layer ${parsedLayer.id} to active layers`, layerObject],
+                })
+            }
+            return layerObject
+        })
+        .filter((layer) => !!layer)
+
+    layersStore.setLayers(layers, STORE_DISPATCHER_ROUTER_PLUGIN)
+
+    if (featuresRequests.length > 0) {
+        promisesForAllDispatch.push(getAndDispatchFeatures(to, featuresRequests))
+    }
+
+    Promise.all(promisesForAllDispatch).catch((error) => {
+        log.error({
+            title: 'Layers URL param / dispatchLayersFromUrlIntoStore',
+            titleColor: LogPreDefinedColor.Sky,
+            messages: [`Error while processing features in feature preselection.`, error],
+        })
+    })
+}
+
+async function getAndDispatchFeatures(
+    to: vueRouter.RouteLocation,
+    featuresPromise: Promise<LayerFeature>[]
+): Promise<void> {
+    const featuresStore = useFeaturesStore()
+    const positionStore = usePositionStore()
+
+    try {
+        const responses = await Promise.allSettled(featuresPromise)
+        const features = responses
+            .filter((response) => response.status === 'fulfilled')
+            .map((response) => response.value)
+        if (features.length > 0) {
+            featuresStore.setSelectedFeatures({ features }, STORE_DISPATCHER_ROUTER_PLUGIN)
+
+            const extent = getExtentOfGeometries(
+                features.map((feature) => feature.geometry).filter((geometry) => !!geometry)
+            )
+            if (!extent) {
+                return
+            }
+            // If the zoom level has been specifically set to a level, we don't want to override that.
+            // otherwise, we go to the zoom level which encompass all features
+            const query = to.query
+            if (!query.z) {
+                positionStore.zoomToExtent(
+                    {
+                        extent: extent,
+                        maxZoom: positionStore.projection.get1_25000ZoomLevel(),
+                    },
+                    STORE_DISPATCHER_ROUTER_PLUGIN
+                )
+            } else {
+                positionStore.setCenter(
+                    extentUtils.getExtentCenter(extent),
```

Comment:
I check locally, and it gives an error. I think we need to add `getExtentCenter` to `GeoadminExtentUtils` interface.

Reply:
it is, though.
But if you didn't `pnpm run build` the layers package your local "copy" in the other packages was outdated